test/strategy.test.js

```diff
@@ -277,3 +284,42 @@ describe('#authenticate', function (){
     });
   });
 });
+
+describe('#Issuer', function() {
+  it('returns a discover function and well known config object', function() {
+    expect(Issuer.discover).to.be.a('function');
+    expect(Issuer._well_known_config).to.be.an('object');
+  });
+
+  describe('#discover', function() {
+    context('when a url is passed', function() {
+      it('returns a Client contructor and well known config object', function() {
+        Issuer.discover('localhost:5000/').then(function(issuer) {
+          expect(issuer.Client).to.be.a('function');
+        });
+      });
+    });
+
+    context('when a url is not passed', function() {
+      it('rejects with an error', function() {
+        Issuer.discover().catch(function(err) {
+          expect(err).to.be.defined;
+        });
+      });
+    });
+  });
+});
+
+describe('#Client', function() {
+  beforeEach(function (){
+    config = { issuer: 'localhost:5000' };
+  });
+
+  it('returns a new client', function() {
+    Issuer.discover('localhost:5000/').then(function(issuer) {
```

Comment:
you need to return this promise so the test knows to wait for the epects to resolve